Display fiat amount previews in Transaction Details page

rockstardev opened the pull request:
Resolves: #6607

This will be useful now that we have pending transactions and fiat amount can significantly change between when transaction is created versus signed and broadcasted.

https://github.com/user-attachments/assets/66078691-5aeb-49f4-a725-fd49e47e078a

NicolasDorier commented:
Super cool, but need @dstrukt to find better UX. Right now how the user can know it is even clickable?

rockstardev commented:
@NicolasDorier since PR is fully functional let's merge this to get 2.0.7 out, I need it for deployment as people are waiting.

Then we can improve UX in later versions as people have time.

NicolasDorier commented:
Doesn't work 
![image](https://github.com/user-attachments/assets/8bacbd3f-d55e-40f7-bab1-da736fe32f93)

NicolasDorier commented on BTCPayServer/Controllers/UIWalletsController.cs:
```diff
@@ -587,31 +583,43 @@ public async Task<IActionResult> WalletSend(
-                    }
-                }
-                catch (Exception ex) { model.RateError = ex.Message; }
+                var r = await fetchRate(walletId);
```
why this change?  IT doesn't seem you reuse `fetchRate` anywhere?

rockstardev replied:
All the code was in different places, so it made sense to extract it into a single cohesive `fetchRate` method.

NicolasDorier commented on BTCPayServer/Views/UIWallets/_PSBTInfo.cshtml:
```diff
@@ -23,38 +34,36 @@
         @foreach (var input in Model.Inputs)
         {
             <tr>
-                @if (input.Error != null)
```
Why removing errors?

rockstardev replied:
Good catch, vibe coding is to blame. Restored: eec7c374198f6976c6f524c72c2ae6d23b0d534d

NicolasDorier commented on BTCPayServer/Views/UIWallets/_PSBTInfo.cshtml:
```diff
@@ -1,11 +1,22 @@
+    private string ToFiatAmount(string destinationBalance)
+    {
+        var amt = Model.Rate * Decimal.Parse(destinationBalance.Replace("BTC", "").Trim());
+        return amt.ToString("N" + Model.FiatDivisibility) + " " + Model.Fiat;
```
Do not use this to format fiat, we have proper methods to do it. (`DisplayFormatter.Currency`)

NicolasDorier commented on BTCPayServer/Views/UIWallets/_PSBTInfo.cshtml:
```diff
@@ -1,11 +1,22 @@
+@functions {
+    private string ToFiatAmount(string destinationBalance)
+    {
+        var amt = Model.Rate * Decimal.Parse(destinationBalance.Replace("BTC", "").Trim());
```
Any other currency will just crash here

NicolasDorier replied:
Also, I think this also will crash if the OS is not running on US culture.

rockstardev replied:
Fixed: 7de92b0f0f7e9b0a6994814d4ddadc34f281bd4f

rockstardev commented:
@NicolasDorier great review on PR, I've fixed the things you've brought up, ready to merge now.

I've also added a subtle dotted underline hint that fiat value is clickable:

![image](https://github.com/user-attachments/assets/33b27a2e-4ab3-417a-8ac7-d7feeb74ef63)
